distributed/client.py

```diff
@@ -578,14 +597,16 @@ def start(self, asynchronous=None, **kwargs):
                 self._should_close_loop = True
             while not self.loop._running:
                 sleep(0.001)
+
         pc = PeriodicCallback(lambda: None, 1000, io_loop=self.loop)
         self.loop.add_callback(pc.start)
         _set_global_client(self)
-        if asynchronous:
+        self.status = 'connecting'
+
+        if self.asynchronous:
```

Comment:
One issue that arises is that when we deserialize futures or queues or variables we don't know if the client should be asynchronous or synchronous.  The user doesn't have a chance to tell us in this case.